ostd/src/logger.rs

```diff
@@ -48,6 +56,11 @@ impl log::Log for Logger {
             (timestamp, level, record_str)
         };
 
+        // Use a global lock to prevent interleaving of log messages.
+        use crate::sync::SpinLock;
+        static MESSAGE_LOCK: SpinLock<()> = SpinLock::new(());
```

Comment:
```rust
fn log(&self, record: &Record) {
```

Since the function signature has chose to use the term _record_, let's keep using it rather than some other terms like _message_.

```rust
        // Use a global lock to prevent interleaving of log records.
        use crate::sync::SpinLock;
        static RECORD_LOCK: SpinLock<()> = SpinLock::new(());
```